### PR TITLE
chore(storage-browser): integrate location and detail views with comp…

### PR DESCRIPTION
**Pull request by cshfang**
…osable DataTable

#### Description of changes

This PR integrates the updated composable DataTable with the Locations and LocationDetail views.

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x]  PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ashika112 — COMMENTED** [review]
Looks good to me. But now merge conflict 😅

File: packages/e2e/cypress/integration/common/storagebrowser.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { When } from '@badeball/cypress-cucumber-preprocessor';
 When(
   'I drag and drop a file into the storage browser with file name {string}',
   (fileName: string) => {
-    cy.get('[data-testid="storage-browser-table"').trigger('drop', {
+    cy.get('[class="storage-browser__drop-zone"').trigger('drop', {
```
Comment (ashika112) [inline packages/e2e/cypress/integration/common/storagebrowser.ts:6]: nit: is this change needed if u are not running the test atm?

File: packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx [hunk 2]
```diff
@@ -56,27 +55,25 @@ export function LocationsView({
...
+    <ControlsContextProvider
+      data={{
+        isDataRefreshDisabled: isLoading,
+        tableData: getLocationsViewTableData({ pageItems, onNavigate }),
+        searchPlaceholder,
+      }}
+      onSearch={onSearch}
+      onRefresh={onRefresh}
+    >
+      {' '}
```
Comment (ashika112) [inline packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx:76]: nit: remove?

File: packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx [hunk 3]
```diff
@@ -98,7 +95,11 @@ export function LocationsView({
         <LocationsMessage show={hasError} message={message} />
         <Loading show={isLoading} />
         {hasError ? null : (
```
Comment (ashika112) [inline packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx:97]: same comment as above.

File: packages/e2e/cypress/integration/common/storagebrowser.ts (hunk 1 above)
Comment (cshfang) [reply packages/e2e/cypress/integration/common/storagebrowser.ts:6]: It'll eventually need to be fixed and the target will be this still I think

**Review by ashika112 — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/__tests__/LocationDetailView.spec.tsx [hunk 4]
```diff
@@ -1,19 +1,200 @@
...
+  it('loads initial location items for a BUCKET location as expected', () => {
+    useStoreSpy.mockReturnValueOnce([
+      {
+        location: { current: location, path: '', key: location.prefix },
+        locationItems: { fileDataItems: undefined },
+      } as StoreModule.UseStoreState,
+      dispatchStoreAction,
+    ]);
+    render(<LocationDetailView />);
+
```
Comment (pranavosu) [inline packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/__tests__/LocationDetailView.spec.tsx:162]: can we add an assertion to check if the table is rendered?

File: packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts [hunk 5]
```diff
@@ -21,19 +21,27 @@ interface UseLocationDetailView {
   isLoading: boolean;
   isPaginateNextDisabled: boolean;
   isPaginatePreviousDisabled: boolean;
+  currentLocation: LocationData | undefined;
```
Comment (pranavosu) [inline packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts:24]: when will currentLocation be `undefined`?

File: packages/react-storage/src/components/StorageBrowser/controls/hooks/useDataTable.ts [hunk 6]
```diff
@@ -152,12 +152,7 @@ export const useDataTable = (): DataTableProps | null => {
       .flat();
   }, [sortState, tableData]);
```
Comment (pranavosu) [inline packages/react-storage/src/components/StorageBrowser/controls/hooks/useDataTable.ts:153]: possibly unrelated to the PR, but how does refresh affect sort state for the table?

Comment (cshfang) [reply packages/react-storage/src/components/StorageBrowser/controls/hooks/useDataTable.ts:153]: I'm not sure what the right behavior should be but when hitting refresh today, the sort state is maintained

File: packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts (hunk 5 above)
Comment (cshfang) [reply packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts:24]: Theoretically, on the views it is used, never. But it is technically possible for it to be undefined on the location state. Rather than just doing some one size fits all cut off in the hook, keeping this allows the downstream consumers to decide how best to deal it.

**Review by pranavosu — APPROVED** [review]

**Review by ashika112 — APPROVED** [review]
